app/lib/katello/util/cdn_var_substitutor.rb

```diff
@@ -90,21 +90,38 @@ def substituable?(path)
         path.include?("$")
       end
 
-      def valid_substitutions?(contnent_url, substitutions)
-        real_path = gsub_vars(contnent_url, substitutions)
+      def valid_substitutions!(content, substitutions)
```

Comment:
So reading this a bit more:  https://www.ruby-forum.com/topic/176830#773946

from the horses mouth. ! by convention is not reserve purely for modifications, although it may be used as such.  It is meant to be a more destructive version of the non-!  and thus unless you have a non-! method you really shouldn't have ! version